src/Analyzer/FunctionNode.cpp

```diff
@@ -31,6 +32,15 @@ FunctionNode::FunctionNode(String function_name_)
     children[arguments_child_index] = std::make_shared<ListNode>();
 }
 
+const DataTypes & FunctionNode::getExpectedArgumentTypes() const
+{
+    if (!function)
+        throw Exception(ErrorCodes::LOGICAL_ERROR,
```

Comment:
Better to throw `UNSUPPORTED_METHOD` and use `Function is not resolved` as exception message.